src/db/update/Walk.cxx

```diff
@@ -415,21 +415,18 @@ UpdateWalk::DirectoryMakeUriParentChecked(Directory &root,
 	StringView uri(_uri);
 
 	while (true) {
-		auto s = uri.Split('/');
-		const std::string_view name = s.first;
-		const auto rest = s.second;
+		auto [name, rest] = uri.Split('/');
 		if (rest == nullptr)
 			break;
 
 		if (!name.empty()) {
-			directory = DirectoryMakeChildChecked(*directory,
-							      std::string(name).c_str(),
-							      s.first);
+			directory = DirectoryMakeChildChecked(
+				*directory, std::string(name).c_str(), name);
```

Comment:
No change here, just reformatted the code (in a way that violates MPD's coding style).

Reply:
I just ran got clang-tidy

Reply:
Please stop hiding behind the tools you've chosen to use. It's not the tools' fault that you submit broken code often. Tools are just that - tools. You should manually review everything before you submit it. Everybody can use such a tool, everybody can run clang-tidy. The difficult part is understanding what happens and verify if what happens is good.

Reply:
The goal is that all my PRs go through clang-tidy so as to have a consistent codebase.

I found something strange while playing around with .clang-format.

```
--- a/src/db/update/Walk.cxx
+++ b/src/db/update/Walk.cxx
@@ -432,9 +432,8 @@ UpdateWalk::DirectoryMakeUriParentChecked(Directory &root,
                        break;
 
                if (!name.empty()) {
-                       directory = DirectoryMakeChildChecked(*directory,
-                                                             std::string(name).c_str(),
-                                                             name);
+                       directory = DirectoryMakeChildChecked(
+                               *directory, std::string(name).c_str(), name);
                        if (directory == nullptr)
                                break;
                }
--- a/src/output/plugins/WasapiOutputPlugin.cxx
+++ b/src/output/plugins/WasapiOutputPlugin.cxx
@@ -322,7 +322,9 @@ void WasapiOutput::Enable() {
 
        if (enumerate_devices && SafeTry([this]() { EnumerateDevices(); })) {
                for (const auto &[device, desc] : device_desc) {
-                       FormatNotice(wasapi_output_domain, "Device \"%u\" \"%s\"", device,
+                       FormatNotice(wasapi_output_domain,
+                                    "Device \"%u\" \"%s\"",
+                                    device,
                                     desc.c_str());
                }
        }

```

That second one was caused by setting BinPackParameters to false.

As far as I can tell. clang-tidy works properly with calling functions directly but not when assigning them.